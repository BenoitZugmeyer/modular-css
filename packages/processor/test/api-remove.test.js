"use strict";

const namer = require("@modular-css/test-utils/namer.js");
const relative = require("@modular-css/test-utils/relative.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".remove()", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });
            
        it("should remove a relative file", async () => {
            await processor.string(
                "./simple.css",
                ".wooga { }"
            );

            processor.remove("./simple.css");

            expect(relative(processor.dependencies())).toMatchSnapshot();
        });

        it("should remove an absolute file", async () => {
            await processor.string(
                "./packages/processor/test/specimens/simple.css",
                ".wooga { }"
            );

            processor.remove(require.resolve("./specimens/simple.css"));

            expect(relative(processor.dependencies())).toMatchSnapshot();
        });

        it("should remove multiple files", async () => {
            await processor.string("./a.css", ".a { }");
            await processor.string("./b.css", ".b { }");
            await processor.string("./c.css", ".c { }");

            processor.remove([
                "./a.css",
                "./b.css",
            ]);

            expect(relative(processor.dependencies())).toMatchSnapshot();
        });

        it("should return an array of removed files", async () => {
            await processor.string("./a.css", ".a { }");
            await processor.string("./b.css", ".b { }");
            await processor.string("./c.css", ".c { }");

            expect(
                relative(
                    processor.remove([
                        "./a.css",
                        "./b.css",
                    ])
                )
            ).toMatchSnapshot();
        });
    });
});
