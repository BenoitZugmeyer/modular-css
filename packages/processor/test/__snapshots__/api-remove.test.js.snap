// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .remove() should remove a relative file 1`] = `Array []`;

exports[`/processor.js .remove() should remove an absolute file 1`] = `Array []`;

exports[`/processor.js .remove() should remove multiple files 1`] = `
Array [
  "c.css",
]
`;

exports[`/processor.js .remove() should return an array of removed files 1`] = `
Array [
  "a.css",
  "b.css",
]
`;
